backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/commissioninstance/services/CommissionConfigFactory.java

```diff
@@ -112,20 +129,66 @@ public CommissionConfigFactory(@NonNull final CommissionConfigStagingDataService
 				.orgId(contractRequest.getOrgId())
 				.dateOrdered(contractRequest.getCommissionDate())
 				.build();
-		final ImmutableList<I_C_Flatrate_Term> commissionTermRecords = flatrateDAO
+		final ImmutableList.Builder<I_C_Flatrate_Term> commissionTermRecordsBuilder = ImmutableList.builder();
+
+		flatrateDAO
 				.retrieveTerms(termsQuery)
 				.stream()
 				.filter(termRecord -> CommissionConstants.TYPE_CONDITIONS_COMMISSION.equals(termRecord.getType_Conditions()))
-				.collect(ImmutableList.toImmutableList());
+				.forEach(commissionTermRecordsBuilder::add);
+
+		getConditionsForSalesRepWithoutContract()
+				.map(flatrateConditions -> createGenericContract(flatrateConditions, commissionTermRecordsBuilder.build(), allBPartnerIds, beneficiary))
+				.ifPresent(contractTermList -> contractTermList.forEach(commissionTermRecordsBuilder::add));
 
 		final ImmutableMap<FlatrateTermId, CommissionConfig> contractId2Config = createCommissionConfigsFor(
-				commissionTermRecords,
+				commissionTermRecordsBuilder.build(),
 				contractRequest.getCustomerBPartnerId(),
 				contractRequest.getSalesProductId());
 
 		return CollectionUtils.extractDistinctElements(contractId2Config.values(), Function.identity());
 	}
 
+	@NonNull
+	private ImmutableList<I_C_Flatrate_Term> createGenericContract(
+			@NonNull final I_C_Flatrate_Conditions flatrateCondition,
+			@NonNull final ImmutableList<I_C_Flatrate_Term> commissionTermRecords,
+			@NonNull final ImmutableList<BPartnerId> hierarchyBPartnerIds,
+			@NonNull final Beneficiary endCustomer)
+	{
+		final ImmutableSet<Integer> salesRepIdsWithContract = commissionTermRecords.stream()
+				.map(I_C_Flatrate_Term::getBill_BPartner_ID)
+				.collect(ImmutableSet.toImmutableSet());
+
+		final ImmutableSet<BPartnerId> salesRepIdsWithoutContract = hierarchyBPartnerIds.stream()
+				.filter(salesRepId -> !salesRepIdsWithContract.contains(salesRepId.getRepoId()))
+				.filter(salesRepId -> salesRepId.getRepoId() != endCustomer.getBPartnerId().getRepoId())
+				.collect(ImmutableSet.toImmutableSet());
+
+		if (salesRepIdsWithoutContract.isEmpty())
+		{
+			return ImmutableList.of();
+		}
+
+		final List<I_C_BPartner> bPartnersWithoutContracts = bPartnerDAO.retrieveByIds(salesRepIdsWithoutContract);
+
+		final ProductId commissionProductId = commissionProductService.getCommissionProduct(ConditionsId.ofRepoId(flatrateCondition.getC_Flatrate_Conditions_ID()));
+
+		final I_M_Product commissionProductRecord = productDAO.getById(commissionProductId);
+
+		final FlatrateTermCreator termCreator = FlatrateTermCreator.builder()
+				.ctx(Env.getCtx())
+				.products(ImmutableList.of(commissionProductRecord))
+				.startDate(Timestamp.from(Instant.now()))
+				.endDate(Timestamp.from(Instant.now().plus(NO_COMMISSION_AGREEMENT_DEFAULT_CONTRACT_DURATION)))
+				.conditions(flatrateCondition)
+				.bPartners(bPartnersWithoutContracts)
+				.isCompleteDocument(true)
+				.build();
+
```

Comment:
please log this to `Loggables.withLogger(Level.Info)..`

Reply:
done